### PR TITLE
dev: update hack locally details for starting a REPL

### PR DESCRIPTION
**Pull request by practicalli-john**
📓 Description

Update hacking locally instructions when starting a repl, highlighing a dependency on [Practicalli Clojure CLI Config](https://practical.li/clojure/clojure-cli/practicalli-config/) or using `clj` as a basic alternative

- add `:dev/env` alias to `deps.ed` to include `dev` and `test` paths, extra dependencies for `portal` and `tools.namespace`

- update `README.md` with `clj` command details

:octocat: Type of change

_Please tick `x` relevant options, delete those not relevant_

- [ ] New feature
- [ ] Deprecate feature
- [ ] Development workflow
- [X] Documentation
- [ ] Continuous integration workflow

:beetle: How Has This Been Tested?

- [ ] unit test
- [X] linter check
- [X] GitHub Action checkers

:eyes: Checklist

- [X] Code follows the [Practicalli cljstyle configuration](https://practical.li/clojure/clojure-cli/clojure-style/#cljstyle)
- [X] Add / update alias docs and README where relevant
- [X] Changelog entry describing notable changes
- [x] Request maintainers review the PR

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,7 +50,9 @@ After installing The Arqivist in Slack, you can interact with the bot in two way
 ## Hacking locally
 
 * Start a Postgres instance with `docker compose up -d postgres`
-* Start a REPL with `make repl`
+* Start a REPL with either:
+    * `make repl` rich REPL prompt (requireds [Practicalli Clojure CLI Config](https://practical.li/clojure/clojure-cli/practicalli-config/))
```
Comment (jcpsantiago) [inline README.md:54]: typo in requires :)

**Review by jcpsantiago — CHANGES_REQUESTED** [review]
found a typo 🗡️

File: README.md (hunk 1 above)
Comment (practicalli-john) [reply README.md]: Ooops, fixed :)

**Review by jcpsantiago — APPROVED** [review]
🚀

**Pull request merged**